core/src/uix/compiler/attributes.cljs

```diff
@@ -0,0 +1,180 @@
+    (map? x) (reduce-kv custom-kv-conv #js {} x)
+    (coll? x) (clj->js x)
+    (ifn? x) #(apply x %&)
+    :else (clj->js x)))
+
+(defn convert-prop-value-shallow [x]
+  (if (map? x)
+    (reduce-kv kv-conv-shallow #js {} x)
+    x))
+
+(defn class-names-coll [class]
+  (let [^js/Array classes (reduce (fn [^js/Array a c]
+                                    (when ^boolean c
+                                      (->> (if (keyword? c) (-name ^not-native c) c)
+                                           (.push a)))
+                                    a)
+                                  #js []
+                                  class)]
+    (when (pos? (.-length classes))
+      (.join classes " "))))
+
+(defn class-names-map [class]
+  (let [^js/Array classes (reduce-kv (fn [^js/Array a b ^boolean c]
+                                       (when c
+                                         (->> (if (keyword? b) (-name ^not-native b) b)
+                                              (.push a)))
+                                       a)
+                                     #js []
+                                     class)]
+    (when (pos? (.-length classes))
+      (.join classes " "))))
+
+(defn ^string class-names
+  ([])
+  ([class]
+   (cond
+     ;; {c1 true c2 false}
+     (map? class)
+     (class-names-map class)
+
+     ;; [c1 c2 c3]
+     (or (array? class) (coll? class))
+     (class-names-coll class)
+
+     ;; :c1
+     (keyword? class)
+     (-name ^not-native class)
+
+     :else class))
+  ([a b]
+   (if ^boolean a
+     (if ^boolean b
+       (str (class-names a) " " (class-names b))
+       (class-names a))
+     (class-names b)))
+  ([a b & rst]
+   (reduce class-names (class-names a b) rst)))
+
+(defn set-id-class
+  "Takes the id and class from tag keyword, and adds them to the
+  other props. Parsed tag is JS object with :id and :class properties."
+  [props id-class]
+  (let [id (aget id-class 1)
+        classes ^js/Array (aget id-class 2)]
+    (cond-> props
+            ;; Only use ID from tag keyword if no :id in props already
+            (and (some? id) (nil? (get props :id)))
+            (assoc :id id)
+
+            ;; Merge classes
+            (and (some? classes) (pos? (.-length classes)))
+            (assoc :class (class-names classes (get props :class))))))
+
+(defn convert-props [props id-class ^boolean shallow?]
+  (let [class (get props :class)
+        props (-> props
+                  (cond-> class (assoc :class (class-names class)))
+                  (set-id-class id-class))]
+    (cond
+      ^boolean (aget id-class 3)
+      (convert-custom-prop-value props)
+
+      shallow?
+      (convert-prop-value-shallow props)
+
+      :else (convert-prop-value props))))
+
+(defn interpret-attrs [attrs id-class shallow?]
+  (if (map? attrs)
+    (convert-props attrs id-class shallow?)
+    (do
+      (js/console.error "UIx: a map of DOM attributes is exppected, but got " attrs)
```

Comment:
Not a biggie but if this error can appear in React Native, it'd be misleading.

"A map of element attributes" is a good alternative.

Reply:
Good point. Updated.